cloud/ec2_price.go

```diff
@@ -128,9 +139,73 @@ func (cpf *cachingPriceFetcher) cacheEc2Prices() error {
 			}] = price
 		}
 	}
+	grip.Debug("Finished parsing on-demand pricing")
 	return nil
 }
 
+func (cpf *cachingPriceFetcher) getLatestLowestSpotCostForInstance(client AWSClient, settings *NewEC2ProviderSettings, os osType) (float64, error) {
+	osName := string(os)
+	if settings.IsVpc {
+		osName += " (Amazon VPC)"
+	}
+	grip.Debug(message.Fields{
+		"message":       "getting spot history",
+		"instance_type": settings.InstanceType,
+		"os_name":       osName,
+	})
+	prices, err := client.DescribeSpotPriceHistory(&ec2.DescribeSpotPriceHistoryInput{
+		// passing a future start time gets the latest price only
+		StartTime:           makeTimePtr(time.Now().UTC().Add(24 * time.Hour)),
+		InstanceTypes:       []*string{makeStringPtr(settings.InstanceType)},
+		ProductDescriptions: []*string{makeStringPtr(osName)},
+	})
+	if err != nil {
+		return 0, errors.Wrap(err, "error getting spot price history")
+	}
+	if len(prices.SpotPriceHistory) == 0 {
+		return 0, errors.New("no prices found")
+	}
+	var min float64
+	for i := range prices.SpotPriceHistory {
+		p, err := strconv.ParseFloat(*prices.SpotPriceHistory[i].SpotPrice, 0)
+		if err != nil {
+			return 0, errors.Wrap(err, "error parsing spot price")
+		}
+		if min == 0 || p < min {
+			min = p
+		}
+	}
+	return min, nil
+}
+
+func (m *ec2Manager) getProvider(h *host.Host, ec2settings *NewEC2ProviderSettings) (ec2ProviderType, error) {
+	if m.provider == spotProvider {
+		return spotProvider, nil
+	}
+	if m.provider == onDemandProvider {
```

Comment:
shouldn't these consts' be in the global package, or use them?

Reply:
No, these are specific to this implementation, are not used anywhere else, and are an iota.